contrib/git-hooks/pre-commit

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2018 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+#
+# This pre-commit hook checks formatting for each commit.
+#
+
+#
+# If set to true, committing is aborted if the files are not formatted
+# correctly. If set to false, files will be reformatted automatically.
+#
+REPORT_ONLY=true
+
+basedir="$(git rev-parse --show-toplevel 2>/dev/null)"
+
+if [ ! -f "${basedir}"/contrib/utilities/indent-branch ]; then
+  echo "*** This script must be run from within the deal.II git repository."
+  exit 1
+fi
+
+cd "${basedir}"
+"${basedir}"/contrib/utilities/indent-branch
```

Comment:
@masterleinad This would be the new pre-commit hook. I will push a commit for the `REPORT_ONLY` option in a minute.